dask/dataframe/tests/test_dataframe.py

```diff
@@ -170,6 +170,38 @@ def test_set_index_interpolate():
     assert d2.divisions[3] == 2.
 
 
+def test_set_index_interpolate_int():
+    L = sorted(list(range(0, 200, 10))*2)
+    df = pd.DataFrame({'x': 2*L})
+    d = dd.from_pandas(df, 2)
+    d1 = d.set_index('x', npartitions=10)
+    assert all(np.issubdtype(type(x), np.integer) for x in d1.divisions)
+    assert any(x % 10 != 0 for x in d1.divisions)
```

Comment:
Do we want this to be the case?  If someone made a solution that always landed on actual data values would that be wrong?